File: c/udemy/4-condicional/baskara.c
// Fazer um programa para ler os três coeficientes de uma equação do segundo grau. Usando a fórmula de Bhaskara, calcular e mostrar os valores das raízes x1 e x2 da equação com quatro casas decimais, conforme exemplo. Se a equação não possuir raízes reais, mostrar uma mensagem

#include <stdio.h>
#include <math.h>

int main(){

    double a, b, c, delta, x1, x2;

    printf("Valor de A: ");
    scanf("%lf", &a);
    printf("Valor de B: ");
    scanf("%lf", &b);
    printf("Valor de C: ");
    scanf("%lf", &c);

    delta = pow(b, 2) - (4 * a * c);
    
    if (delta > 0){
        x1 = (-b + sqrt(delta))/ (2 * a);
        x2 = (-b - sqrt(delta))/ (2 * a);
        printf("\nPrimeira raiz: %.4lf", x1);
        printf("\nSegunda raiz: %.4lf", x2);
    }
    else if (delta == 0){
        x1 = (-b + sqrt(delta))/ (2 * a);
        printf("\nRaiz real única: %.4lf", x1);
    }
    else{
        printf("\nSem raiz real");
    }

    return 0;

}

File: python/udemy/4-estrutura-condicional/baskara.py
# Fazer um programa para ler os três coeficientes de uma equação do segundo grau. Usando a fórmula de Baskara, calcular e mostrar os valores das raízes x1 e x2 da equação com quatro casas decimais, conforme exemplo. Se a equação não possuir raízes reais, mostrar uma mensagem.

import math

a = float(input('coeficiente A: '))
b = float(input('coeficiente B: '))
c = float(input('coeficiente C: '))

delta = (b**2) - (4*a*c)

if a == 0 or delta < 0:
    print('A equação não possui raízes reais.')

elif delta > 0:
    x1 = (-b + math.sqrt(delta))/ (2 * a)
    x2 = (-b - math.sqrt(delta))/ (2 * a)
    print(f'Primeira raiz real: {round(x1, 4)}')
    print(f'Segunda raiz real: {round(x2, 4)}')

elif delta == 0:
    x = (-b + math.sqrt(delta))/2
    print(f'Raiz real única: {round(x, 4)}')    